src/clojure/nrepl/middleware.clj

```diff
@@ -178,3 +178,18 @@
        extend-deps
        (topologically-sort comparator)
        (map :implemented-by)))
+
+;; Registering dynvars that are used to configure middleware. This lives here
+;; and not in nrepl.middleware.session to avoid circular dependencies.
+
+(def explicit-per-session-dynvars
+  "Mapping of dynamic variables that always have to be established for a new
+  session to their initial values."
+  (atom {}))
+
+(defn register-per-session-dynvars
```

Comment:
I'm wondering if it won't be a bit nice if we moved the dynvars registration into `set-descriptor`, so that we can just list the dynvars in the middleware descriptors and include them automatically in the middleware docs.

Reply:
Would it be OK for you that `set-descriptor!` performs a mutable side-effect unrelated to the middleware object itself?